src/routing/tests/mod.rs

```diff
@@ -1493,7 +1495,9 @@ async fn handle_elders_update() -> Result<()> {
         event_rx.recv().await,
         Some(Event::EldersChanged { elders, .. }) => {
             assert_eq!(elders.key, pk1);
-            assert_eq!(elders.elders, elder_names1);
+            assert!(elders.added.iter().all(|a| elder_names1.contains(a)));
+            assert!(elders.existing.iter().all(|a| elder_names1.contains(a)));
```

Comment:
`assert_eq!(elder_names1, elders.added.union(&elders.existing))` shall be enough and more accurate for the above two `all` check?